### PR TITLE
[SPARK-12566] [SPARK-14324] [ML] [WIP] GLM model family, link function support in SparkR:::glm

### PR DESCRIPTION
**Pull request by yanboliang**
## What changes were proposed in this pull request?
- SparkR glm supports families and link functions which match R's signature for family.
- SparkR glm API refactor. The comparative standard of the new API is R glm, so I only expose the arguments that R glm supports: `formula, family, data, epsilon and maxit`.
- This PR is focus on glm() and predict(), summary statistics will be done in a separate PR after this get in.
- This PR depends on #12287 which make GLMs support link prediction at Scala side. After that merged, I will add more tests for predict() to this PR.
## How was this patch tested?

Unit tests.

cc @mengxr @jkbradley @hhbyyh

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/ml/r/GeneralizedLinearRegressionWrapper.scala [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    val features = featureAttrs.map(_.name.get)
+    // assemble and fit the pipeline
+    val glm = new GeneralizedLinearRegression()
+      .setFamily(family)
+      .setLink(link)
+      .setFitIntercept(rFormula.hasIntercept)
+      .setTol(epsilon)
+      .setMaxIter(maxit)
+    val pipeline = new Pipeline()
+      .setStages(Array(rFormula, glm))
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/ml/r/GeneralizedLinearRegressionWrapper.scala]: We can use `rFormulaModel` to avoid fitting again.

**Comment by mengxr** [comment]
LGTM except one place where we can avoid re-fitting. @yanboliang Could you create follow-up JIRAs so we can add the missing feature back? Thanks!

**Comment by yanboliang** [comment]
@mengxr We already have [SPARK-13925](https://issues.apache.org/jira/browse/SPARK-13925) for add  summary statistics for SparkR glm. I will send a PR after this get in.

**Comment by mengxr** [comment]
LGTM. Merged into master. Thanks!

**Pull request closed without merging**